github/resource_github_membership.go

```diff
@@ -126,8 +132,19 @@ func resourceGithubMembershipDelete(d *schema.ResourceData, meta interface{}) er
 	orgName := meta.(*Owner).name
 	ctx := context.WithValue(context.Background(), ctxId, d.Id())
 
-	_, err = client.Organizations.RemoveOrgMembership(ctx,
-		d.Get("username").(string), orgName)
+	username := d.Get("username").(string)
+	downgradeOnDestroy := d.Get("downgrade_on_destroy").(bool)
+	downgradeTo := "member"
+
+	if downgradeOnDestroy {
+		log.Printf("[INFO] Downgrading '%s' membership for '%s' to '%s'", orgName, username, downgradeTo)
+		_, _, err = client.Organizations.EditOrgMembership(ctx, username, orgName, &github.Membership{
```

Comment:
What happens if a "member" is destroyed while `downgrade_on_destroy` is true?

Reply:
If a membership was revoked outside of Terraform and then afterward removed from Terraform, the code at the time you commented would have re-invited the user back to the organization.

Now I load the membership status and check if the user is still in the org before I call `EditOrgMembership`.

Thanks.